modules/kind/kindcluster.go

```diff
@@ -71,8 +71,43 @@ func (c *Cluster) ID() string {
 	return fmt.Sprintf("pkg.package-operator.run/cardboard/modules/kind.Cluster{name:%s}", c.name)
 }
 
+func (c *Cluster) IPv4() (string, error) {
```

Comment:
I think the function signature and/or comment should highlight the fact that this gives you the IPv4 address of the *first control plane node* it encounters.

There are some `TODO: augment error` comments that I think should be adressed as well :)

Reply:
I was thinking about this function. The filtering is very bespoke to the one use case we have so we should probably make it clear (via naming etc) that this function is supposed to help us solve our use case of finding the node with the registry on it or if we want to keep it generic.

If we want to keep it generic, maybe one could pass in their own selector/filter function and this package could then export a pre-defined `FirstControlPlaneNode` selector function?

Reply:
I renamed the function to `ControlPlaneIPv4`, added a comment to explain the specific behavior and removed the `TODO: augment error` comments since other errors are not usually augmented

Reply:
> I renamed the function to `ControlPlaneIPv4`

This still hides the fact that this function returns the ip address of the *first* control-plane node it encounters.

> I renamed the function to ControlPlaneIPv4, added a comment to explain the specific behavior and removed the TODO: augment error comments since other errors are not usually augmented

:(